nodes/src/nodes/rerank_cohere/rerank_client.py

```
@@ -0,0 +1,181 @@
+        Args:
+            logicalType: The logical type of the node.
+            config: Configuration dictionary containing API key, model, etc.
+            bag: Shared bag for cross-node communication.
+        """
+        self._logicalType = logicalType
+        self._model = config.get('model', 'rerank-v3.5')
+        self._top_n = config.get('top_n', 5)
+        self._min_score = config.get('min_score', 0.0)
+
+        if self._top_n < 1:
+            raise ValueError(f'top_n must be >= 1, got {self._top_n}')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)